include/k4ainternal/global.h

```diff
@@ -41,7 +47,7 @@ void global_init_once(k4a_init_once_t *init_once, k4a_init_once_function_t *init
     static _global_type_ _##_global_type_##_private;                                                                   \
     static void fn_##_global_type_##_init_function(void)                                                               \
     {                                                                                                                  \
-        memset(&_##_global_type_##_private, 0, sizeof(_##_global_type_##_private));                                    \
+        DEFAULT_INIT(_global_type_, _##_global_type_##_private);                                                       \
```

Comment:
`K4A_DECLARE_GLOBAL` is used with `logger_global_context_t`, which contains C++ types, and should not be memset.

Warning: `clearing an object of type ‘struct logger_global_context_t’ with no trivial copy-assignment; use assignment or value-initialization instead`